misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -98,8 +98,11 @@ public ModuleCoverage generateReport(JarResolver jarResolver) throws IOException
             // For each jar file found, we unzip it for this particular module
             for (Path jarPath : filteredPathList) {
                 try {
-                    // Creates coverage folder with each class per module
-                    CodeCoverageUtils.unzipCompiledSource(jarPath, coverageDir, orgName, packageName, version);
+                    // Filters out only the jars belonging to the current project. Excludes external jars
+                    if (jarPath.toString().contains(orgName + "/" + packageName + "/" + version)) {
```

Comment:
The jarPath is in a filesystem path as I see. In that case, this would fail in windows. We should use the separator in an OS-independent way. e.g `File.separator`

Reply:
Shall we try to use `target.cachesPath()` so that we don't have to hardcode the path?

Reply:
Changed the filter to include `target.cachesPath()` rather than a hardcoded path